### DISCUSSION TITLE
RestEasy 6,0.1 and RESTEASY003215: could not find writer for content-type application/json

### DISCUSSION THREAD

**Discussion by mikepoz59** [resteasy/resteasy#3293] [General]
Trying to use RestEasy for a Java 8 project that makes a simple API Post, and I keep getting the above error and I can't seem to make it go away.

My POM dependencies are: 
- resteasy-core
- resteasy-client
- resteasy-jackson2-provider

My code is simple:
client = (ResteasyClient) ClientBuilder.newBuilder()
                .trustStore(keyStore).build();
        ResteasyWebTarget target = client.target(authUrl);
        return target.proxy(AuthInterface.class);
        
When I invoke 
service.auth(Credentials)
where credentials is a populated POJO with three fields (String, String, and Boolean)  I get the error in the discussion title.

I've tried everything I can find on the internet, including the often suggested adding of Jackson dependencies, but nothing works.

Note that as a Java 11 project this works, however we need this to be a Java 8 component, and from what I found in the source code, this version of RestEasy has this in the POM which seems to indicate that it should work with Java 8:

        
        <jdk.min.version>11</jdk.min.version>
        <javadoc.additional.params>--release=8</javadoc.additional.params>
        
Am I interpreting this wrong? I have searched high and low but outside of going to each release and looking at the RestEasy POM.xml, it seems that it's somewhere in the 4.x range of releases.  Can anyone tell me the latest version that is Java 8 compatible?

Thanks!
Mike Poz

**Comment by jamezp** [comment]
Hi Mike,
The latest version of RESTEasy that supports Java 8 and Jakarta REST 2.1 would be RESTEasy 5.0.4.Final. The 4.x releases should too, but I would suggest 5.0.4.Final since 5.x is going to be the supported Jakarta REST 2.1 version.

It does look like what you are doing should work. What version of RESTEasy are you trying?

**Reply by mikepoz59** [reply]
Hey Jamez, 6.0.1-RELEASE, I'll step my dependency down to 5.0.4 and try that.

Thank you!

Mike Poz

**Reply by jamezp** [reply]
Ah yes, 6.0.1.Final is Jakarta REST 3.0 so it would use the `jakarta.ws.rs.*` namespace. If that is acceptable, I'd suggest using 6.2.0.Final which targets Jakarta REST 3.1. If not, then 5.0.4.Final is the way to go :)

**Reply by jamezp** [reply]
Wait, 6.2.0.Final would require Java 11. Sorry, I forgot about that part :)
